pyquil/api/qpu.py

```diff
@@ -177,6 +177,7 @@ def run(self, quil_program, classical_addresses=None, trials=1, needs_compilatio
         :return: A list of a list of classical registers (each register contains a bit)
         :rtype: list
         """
+        errors.EmptyProgramError(quil_program)
```

Comment:
This seems like an unconventional way to check for, and potentially raise, an error type, as it reads like it is being raised unconditionally. I would have imagined that you'd create the exception type above via a subclass (`class EmptyProgramError(Exception)`), and then here, and elsewhere, check against the Program being non-empty. @stevenheidel @mpharrigan thoughts?

Reply:
Thanks for the constructive feedback. I will get on them right away. Should I change this to  `errors.CheckIfEmptyProgram(quil_program)` or something of that nature?

Reply:
Ah. I see what you mean.

Reply:
Since the checking logic is just a line or two, my suggestion would be to write it where it's needed in `qpu.py` and `qvm.py`, and reserve the error subclass to just the error message itself:

```
if not quil_program:
    raise EmptyProgramError("You have attempted to run an empty program. Please provide gates or measure instructions to your program...")
```

If that's a too verbose to write multiple times, then I'd make a `check_if_empty_program` function in `pyquil.quil` that is the code above, and call that function for each use.